src/main/java/com/onarandombox/MultiverseCore/commandtools/flags/FlagsManager.java

```diff
@@ -0,0 +1,172 @@
+package com.onarandombox.MultiverseCore.commandtools.flags;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import co.aikar.commands.InvalidCommandArgument;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Manages all the flag groups and parsing.
+ */
+public class FlagsManager {
```

Comment:
Kind of wondering if we should prefix all these flag things with `Command`... `CommandFlagManager` etc.

Reply:
yea actually that sounds better